com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -222,12 +222,18 @@ internal void __endSendClientRpc(NetworkSerializer serializer, uint rpcMethodId,
 #if DEVELOPMENT_BUILD || UNITY_EDITOR
             if (NetworkManager.__rpc_name_table.TryGetValue(rpcMethodId, out var rpcMethodName))
             {
-                NetworkManager.NetworkMetrics.TrackRpcSent(
-                    NetworkManager.ConnectedClients.Select(x => x.Key).ToArray(),
-                    NetworkObjectId,
-                    rpcMethodName,
-                    __getTypeName(),
-                    messageSize);
+                foreach (var client in NetworkManager.ConnectedClients)
+                {
+                    var bytesReported = NetworkManager.LocalClientId == client.Key
+                        ? 0
+                        : messageSize;
+                    NetworkManager.NetworkMetrics.TrackRpcSent(
+                        client.Key,
+                        NetworkObjectId,
+                        rpcMethodName,
+                        __getTypeName(),
+                        bytesReported);
```

Comment:
This pattern of computing `bytesReported` is repeated so often that personally I wouldn't want to see this PR merged until this step has been included in some other function. To be honest, I think this work should maybe be done in `NetworkMetrics`, so that at least all of this is taken care of in one place.